### PR TITLE
The Social Pyramid

### PR DESCRIPTION
**Pull request by Gyrandola**
# Social Pyramid
Introducing the Social Pyramid, a subsystem that allows the current ruler to dynamically change the status of races mid-round, by interacting with the Throne.

This obviously affects the late-joinability of jobs for each race, but not all jobs are affected by this dynamic. For example: bandits, migrants, foreigners, some of the clergy and mercenaries are outside the Social Pyramid and immune to its changes.

https://github.com/user-attachments/assets/af8a4c58-5afe-46cf-9503-6b806d2e7071

WIP

### CODE REVIEW

**Comment by NikolayByondPlayer** [comment]
Advanced racial theory time

**Comment by DelineFortune** [comment]
The church is based on its own ranks and faith
Isnt it supposed to be outside the Social Pyramid too

**Comment by Gyrandola** [comment]
> The church is based on its own ranks and faith Isnt it supposed to be outside the Social Pyramid too

Actually I will probably leave this to the lore maintainers to decide

**Comment by Kevak** [comment]
Mageprentice need an exception for kobolds/verminfolk. Magic cares not for whom holds the talent, only that the talent is there. There's an argument to be made for full on allowing any races to be mageprentices with the _exception_ of horcs/gobbos/orcs not being allowed on account of them being idiots/graggarspawn respectively.

**Comment by Xaytan** [comment]
So there's now no version of smith that is playable for Shunned races?

Can't even play Towner or a refugee version? Prospector isn't a smith. It doesn't get tongs.

As one of the few Towner mains, one of the few smith mains, and someone who happens to play a tiefling... I guess I'm just SOL?

**Comment by Gyrandola** [comment]
> So there's now no version of smith that is playable for Shunned races?
> 
> Can't even play Towner or a refugee version? Prospector isn't a smith. It doesn't get tongs. You can't even buy tongs from the merchant.
> 
> As one of the few Towner mains, one of the few smith mains, and someone who happens to play a tiefling... I guess I'm just SOL?

https://github.com/Rotwood-Vale/Ratwood-Keep/pull/2015

Btw, you can still play smith apprentice as a shunned race too.

**Comment by AukariBunn** [comment]
A coal update
The concept of the social pyramid is quite good, but the race lists refactor is terrible. We actually needed to loose the lists to have a higher player pool for important roles, not to tighten it further. You cant even play towner as some roles, this update quite literally breaks the game. A lot of people rolled up for a refugee because they literally got no other role to play. 
ShareX_g7eZxPE03d
fx1hVD8EJQ
image
image
image

**Comment by AukariBunn** [comment]
> Btw, you can still play smith apprentice as a shunned race too.

Do shunned races just magically disappear once they finish their learning? In fact all non-noble/non-governing roles must be allowed for the most of the races except some like full orcs or something. Especially the roles where your skill and experience matter first like the magos or the physician. And towner being just for 5 races? This is like straight up saying "we dont want you to play town roles, you either play refugee or go fuck yourself and play on dreamkeep or AP"

**Comment by AukariBunn** [comment]
The crew manifest 20 minutes into the round with 101 pop
1 knight, no priest, no merchant/artificer/architect, 1 smith, no soilsons, 2 towners, 17 refugees
This update literally breaks the game. Not the shitcode, but literally making the players not to play.
divtvu9nnP
image

**Comment by AukariBunn** [comment]
Live people's reaction
ShareX_EcKBJ7lOjz

**Comment by Xaytan** [comment]
How to kill your server with one weird trick! Players hate him!

**Comment by ShiningM00N** [comment]
I want to add to Aukari is saying is that the direction this update wanted to enforce is simply not congruent with what the playerbase wants, and if the response to these complaints is "Go somewhere else", the only thing you will do is alienate players in general. There is no "grand purge" of bad players you will accomplish by doing this.

If anything this feels like a massive misstep to the eternal hatred the vision has towards any kind of "adventurer" role, or role that isn't necesarily reliant on the town. It feels like you shot yourself in the foot.

It's fine to make mistakes, and it's fine to fix them.

**Comment by DelineFortune** [comment]
No one is going to type 100000 lines every round
People can barely manage to pay to guards as stewards and now they are supposed to do another job, lmao

Also the update is pointless, because you can have respected race character, take the slot and nothing is happening for you if duke makes you shunned

**Comment by AukariBunn** [comment]
> It's fine to make mistakes, and it's fine to fix them.

If nothing gets changed and this PR gets merged as it is, im making another PR making the new race lists even more loose than the old ones, so there will be drow and vermin dukes.

**Comment by ShiningM00N** [comment]
> Also the update is pointless, because you can have respected race character from roundstart, take the slot and nothing is happening for you if duke makes you shunned

This is a very interesting design flaw because i'm 90% sure that on paper, you thought that this would make things like Half-Orc knights possible, or Drow Knights. However, these are **highly sought after roles that people always go into round start.**

It doesn't matter if the latejoin is changed, because 90% of times, the role is already picked because it's such a sought after role.

**Comment by Alva-Furii** [comment]
In theory, a system for racial hierarchy is good for the setting...

In practice, this PR is kinda stupid as it makes little to no sense in many aspects, ranging from races that can be apprentices but NOT masters to literally not considering if the church should follow the duke's advanced racial theory (it probably should not, church has its own hierarchy and lots of power to just do its own thing) and no issues at all if your race is turned shunned after roundstart while your role requires a respected race.

Idc if I get ignored, just this PR is so raw it still is a living thing.

**Comment by ScienceMom** [comment]
>removes the uncapped pilgrim role and the 20-cap adventurer role and replaces it with a 25-cap frankenstein of both, but with none of the actual choice of either
>"we did this because we want the town to be the main focus"
>REMOVES THE PLAYERS' ABILITY TO PLAY IN THE TOWN
>"if you don't like it leave"

**Comment by notafurryy** [comment]
why are lupians and vulpkians above tabaxi... what the fuck...

**Comment by Zmowlcher** [comment]
This sucks. This is just stupid and ruins the game. The town is populated by many player races and you just removed most of them. Way to go.

**Comment by AukariBunn** [comment]
> This sucks. This is just stupid and ruins the game. The town is populated by many player races and you just removed most of them. Way to go.

"Anthro friendly" my ass

**Comment by AukariBunn** [comment]
> No one is going to type 100000 lines every round People can barely manage to pay to guards as stewards and now they are supposed to do another job, lmao

Dukes can barely manage to purge laws before making new ones, and then there's this

**Comment by Gyrandola** [comment]
Its on me and whoever managed the repo before me for allowing people to slowly chip away at the grim part of the server by allowing shunned races to play every role. One PR with a stupid excuse at a time.
Theres only 5 shunned races out of 20.

 I am sure your level of self insertion hasnt reached the point you cannot respect the setting and spirit of the server at all. But if thats the case, it breaks my heart.

**Comment by DelineFortune** [comment]
Its not even grim part its just another numberfuck content removal for shunned races (third content lock)

Not shunned races have snowflake mercs with op gear, better stats, all jobs

Shunned races have... Refugees (nerfed adventurers) and cutthroat mercs (adventurers 2.0)

Its simply not interesting to play as 10 str shunned race player again packs of 11-14 str furries that have access to combat gear, better skills from start 

The idea is good, without 

"Round-start race changes" part
Ecclesiast for everyone is fine, they are refugee 
Church is separated from the duke they are not supposed to listen duke's racism as well, they do care about fellow followers of faith 
No shunner races in guards under keep control is fine 
Towners part is literally schizo. You can play smith-npc but cannot play the towner.


And im pretty doubt that people are going to use it
Because stewards cannot manage to press buttons to pay guards

**Comment by DelineFortune** [comment]
Also keep in mind if you want to make good racism - give them good stats and it will make the effect

Drows had reputation of shitters before nerf
Orcs have reputation of shitters because they are strong brutes
Tabaxi had a reputation of shitters when they had +3 speed

People that do care about pvp stats can always pick respected lupians for their stats and do shit, this pr will change nothing for them (they can get any role from roundstart anyway)
But this pr kills shunned last races content

**Comment by AukariBunn** [comment]
>or, say, play more than ONE character

People arent playing another character because some schizo decided to cut more than half of the game's content to the race im playing. People play certain characters for a reason. They're gonna just play and contribute to another server where they can actually enjoy the game.

**Comment by Alva-Furii** [comment]
I'm all for the grimdark, racist, hopeless setting Ratwood is _trying_ to be. The issue is, there are literally respected races that _also_ have all the stat bonuses that _also_ can get good gear from the get go. While the shunned... Can't even be a towner, ffs. 

I get it for, like, orcs or goblins, but in reality, jews were universally hated all throughout the medieval and Modern Day Europe yet still were able to work as bankers, traders and other pretty important people. Why? Because there are some kinds of work where it is the skill that comes first, not how respected you are. You don't need to be of a high race to be a blacksmith, mages don't give half a shit about your race as long as you have the gift of Noc and at least a decently wrinkled brain, traders would actually like to have horcs as their goons because these dudes are muscles on top of muscles and don't really need much pay. 

tl;dr: this attempt at racial stuff is stupid and needs lots of refining as it's even less logical than what social stuff took place in the Medieval Times or Renaissance.

**Comment by Gyrandola** [comment]
> I'm all for the grimdark, racist, hopeless setting Ratwood is _trying_ to be. The issue is, there are literally respected races that _also_ have all the stat bonuses that _also_ can get good gear from the get go. While the shunned... Can't even be a towner, ffs. 
> 
> I get it for, like, orcs or goblins, but in reality, jews were universally hated all throughout the medieval and Modern Day Europe yet still were able to work as bankers, traders and other pretty important people. Why? Because there are some kinds of work where it is the skill that comes first, not how respected you are. You don't need to be of a high race to be a blacksmith, mages don't give half a shit about your race as long as you have the gift of Noc and at least a decently wrinkled brain, traders would actually like to have horcs as their goons because these dudes are muscles on top of muscles and don't really need much pay. 
> 
> tl;dr: this attempt at racial stuff is stupid and needs lots of refining as it's even less logical than what social stuff took place in the Medieval Times or Renaissance.

The jew argument doesnt hold because they were persecuted to various degrees, 
See:

**Comment by Alva-Furii** [comment]
> The jew argument doesnt hold because they were persecuted to various degrees,
Alright, rephrasing: even the persecuted jews were able to do different town work, banking and related activities were often the field that was almost exclusively jewish due to many reasons. The current state of this PR literally makes any shunned races unable to do almost any towner activities, which makes for very little content for those who don't wanna make characters of each race just to keep ability to play in town, as most activities happen there and the current policy of the server is to make it more town-centric, which this PR literally fails to at least keep on the same level. I'm not even talking about poorly thought out things like being able to be an apprentice but not a blacksmith/magos or the church with all of its power following all the racial whims of the duke.

**Comment by Gyrandola** [comment]
Putting this on hold until me and the maptainer add city walls and slums, with lesser "town" jobs for shunned races.

**Comment by NikolayByondPlayer** [comment]
> > or, say, play more than ONE character
> 
> People arent playing another character because some schizo decided to cut more than half of the game's content to the race im playing. People play certain characters for a reason. They're gonna just play and contribute to another server where they can actually enjoy the game.

the age of hugbox is over

**Comment by Githubaccount222** [comment]
idk man maybe people would play more than one character if content wasnt being cut constantly

**Comment by Githubaccount222** [comment]
make announcement

**Comment by AukariBunn** [comment]
The Duke Speaks

> make announcement

**Comment by Githubaccount222** [comment]
> The Duke Speaks
> 
> > make announcement

**make announcement**

**Comment by Zmowlcher** [comment]
Okay i have an idea to actually make this work. Make the docks the shunned district. Give the shunned towner role thats just normal towner but for those races. Use those stalls as the actual buisnesses. This keeps everyone within arms reach of each other but very distinctively seperated. I think this would do really well and be very healthy for the server.

**Comment by Keazra** [comment]
I don't care much about how pained people are about the social hierarchy changes but good god the chatlog spam is going to be horrendous from this.

**Comment by Gyrandola** [comment]
> Okay i have an idea to actually make this work. Make the docks the shunned district. Give the shunned towner role thats just normal towner but for those races. Use those stalls as the actual buisnesses. This keeps everyone within arms reach of each other but very distinctively seperated. I think this would do really well and be very healthy for the server.

Not a bad idea. Theres a lot of way slums could be added. 
1) outside city walls
2) In the lower levels, if we have a multi level city
3) In a specific district inside the city

**Comment by Zmowlcher** [comment]
> > Okay i have an idea to actually make this work. Make the docks the shunned district. Give the shunned towner role thats just normal towner but for those races. Use those stalls as the actual buisnesses. This keeps everyone within arms reach of each other but very distinctively seperated. I think this would do really well and be very healthy for the server.
> 
> Not a bad idea. Theres a lot of way slums could be added.
> 
> 1. outside city walls
> 2. In the lower levels, if we have a multi level city
> 3. In a specific district inside the city

My main idea with this was to avoid creating another whole district. I really like the lower levels idea and it would make use of the docks the most. Dividing the docks up between living areas underneath and buisnesses uptop actually makes me wanna be a slum character more than a towner. It gives the captain more people to directly be in charge of you could have extortion rackets going on.

**Comment by Gyrandola** [comment]
Race changes will come in a later PR.

**Comment by Hmeister-real** [comment]
> why are lupians and vulpkians above tabaxi... what the fuck...

They are like the Khajiit

**Comment by Makximus** [comment]
I feel this is a great concept, and punishing an entire race for the actions of a few of their specimen would really set a Grim and Dark tone, but all this is going to culminate in is 100 furry dukes going "HURR DURR,,, HUMAN ARE SHUNNED.... KILL ALL HUMAN... A HUMAN WAS SPOTTED IN FURRY TOWN.... ALERT YOUR LOCAL REICHSKOMISSAR IF YOU SEE ANYTHING..." every round.

races that are already in the respected category by default should not be allowed to be degraded, or alternatively, the duke should get nuked when he does so because Grimoria's multitude of human, elf, dwarf, etc. rulers see their own race be made subhumans and immediately send an army of nine thousand dudes to rape the duke of rockhill to death and then piss all over his court.

**Comment by Gyrandola** [comment]
Test merging the system without any of the round start racw changes.

**Comment by RyszardParys** [comment]
> The crew manifest 20 minutes into the round with 101 pop 1 knight, no priest, no merchant/artificer/architect, 1 smith, no soilsons, 2 towners, 17 refugees This update literally breaks the game. Not the shitcode, but literally making the players not to play. divtvu9nnP image

Because why would a PLAYER decide what i get to play and what i don't get to play? This is dogshit, duh.

**Comment by Gyrandola** [comment]
To be reopened in the future
